build_all/base-libraries/AzureIoTSocket_Ethernet2/src/AzureIoTSocket_Ethernet2.h

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) Microsoft. All rights reserved.
```

Comment:
why is this a `.h` file and not `.c`?

Reply:
It's a .h file because in Arduino you need to include the named library's .h file to include the library.

Reply:
This still doesn’t make sense. It’s a header file inside a `src` directory. So everyone that would include it would have those functions declared and you could have duplicate declarations. Why not have a `.h` file with the public facing functions and include that when needed and have the `.c` with the functions to be linked to? Not quite sure why standard c programming needs to be broken here.

Reply:
I agree, I'm going to move this back into sample_init.cpp like with the esp8266, not sure why Mark did move it here.

Reply:
Done.